libs/chonky/Cargo.toml

```diff
@@ -17,6 +17,9 @@ edition.workspace = true
 error-stack = { workspace = true, public = true }
 
 # Public third-party dependencies
+pdfium-render = { version = "0.8.25", public = true }
+image = { version = "0.24.9", public = true }
+thiserror = { version = "1.0.65", default-features = false }
```

Comment:
We typically store this in the top-level `Cargo.toml` and reference them here with `workspace = true`. Typically, we try to disable as many features as possible by default by setting `default-features = false`, this often results in a smaller sets of crates being required.
We already have `thiserror` listed there, so you can just use 
```toml
thiserror = { workspace = true }
```
The default-features is inheritace from the top-level. 

I don't think any of these libraries needs to be a public dependency as none of them is exposed in the API. By default you can just make them not public, clippy will complain if they should be public.

Reply:
Ok done, I do have to note that since pdfium and image are being returned by functions that are then used by main they do need to be public (for now), once we have a proper examples page we can wrap these implementations so they are not in the public interface.